project-base/storefront/hooks/countries/useCountriesAsSelectOptions.ts

```diff
@@ -0,0 +1,18 @@
+import { useCountriesQueryApi } from 'graphql/generated';
+import { CountryFragmentApi } from 'graphql/generated';
+import { useMemo } from 'react';
+import { SelectOptionType } from 'types/selectOptions';
+
+export const useCountriesAsSelectOptions = () => {
+    const [{ data: countriesData }] = useCountriesQueryApi();
+
+    return useMemo(() => mapCountriesToSelectOptions(countriesData?.countries), [countriesData?.countries]);
+};
+
+const mapCountriesToSelectOptions = (countries: CountryFragmentApi[] | undefined): SelectOptionType[] =>
+    countries?.map((country) => {
+        return {
+            label: country.name,
+            value: country.code,
+        };
+    }) ?? [];
```

Comment:
Maybe simpler 

` countries?.map((country) => ({
            label: country.name,
            value: country.code,
        })
) ?? [];`

would be enough?

Reply:
Good, thanks. Changed.